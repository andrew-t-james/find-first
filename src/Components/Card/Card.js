import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ title, image, location, company, type }) => {
  return (
    <div className="card">
      <header className="card__header">
        <h2 className="card__company">{company}</h2>
        <p className="card__type">{type}</p>
      </header>
      <h3 className="card__title">{title}</h3>
      <section className="card__image--section">
        <img src={image} alt={company}/>
        <p className="card__location">{location}</p>
      </section>
    </div>
  );
};

export default Card;

Card.propTypes = {
  title: PropTypes.string,
  location: PropTypes.string,
  company: PropTypes.string,
  type: PropTypes.string
};