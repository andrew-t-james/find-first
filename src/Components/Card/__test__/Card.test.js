import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card';

describe('<Card />', () => {
  let wrapper;
  const mockJobs = [
    {
      title: 'title',
      id: 2,
      image:'url-here',
      description: 'description',
      location: 'New City',
      type: 'remote'
    }
  ];

  beforeEach(() => wrapper = shallow(<Card {...mockJobs}/>));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
