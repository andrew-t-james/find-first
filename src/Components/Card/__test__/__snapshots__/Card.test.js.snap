// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
<div
  className="card"
>
  <header
    className="card__header"
  >
    <h2
      className="card__company"
    />
    <p
      className="card__type"
    />
  </header>
  <h3
    className="card__title"
  />
  <section
    className="card__image--section"
  >
    <div />
    <p
      className="card__location"
    />
  </section>
</div>
`;
